test/Microsoft.AspNet.Owin.Tests/OwinExtensionTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+
+namespace Microsoft.AspNet.Owin
+{
+    using AppFunc = Func<IDictionary<string, object>, Task>;
+    using CreateMiddleware = Func<
+          Func<IDictionary<string, object>, Task>,
+          Func<IDictionary<string, object>, Task>
+        >;
+    using AddMiddleware = Action<Func<
+          Func<IDictionary<string, object>, Task>,
+          Func<IDictionary<string, object>, Task>
+        >>;
+
+    public class OwinExtensionTests
+    {
+        static AppFunc notFound = async env => env["owin.ResponseStatusCode"] = 404;
+
+        [Fact]
+        public void OwinConfigureServiceProviderAddsServices()
+        {
+            IList<CreateMiddleware> list = new List<CreateMiddleware>();
```

Comment:
Ahh, I wanted to use the Linq's version of `Reverse` which allows me to do the whole thing in one line but I guess I'll separate it and use System.Collection.Generic's version instead.